File: src/main/java/com/sem/keeper/formatter/LocalDateTimeFormatter.java
package com.sem.keeper.formatter;

import org.springframework.context.annotation.Bean;
import org.springframework.format.Formatter;

import java.text.ParseException;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Locale;

/**
 * Saját dátumformátum, mert a default-tól kifolyt a szemem
 */
public class LocalDateTimeFormatter implements Formatter<LocalDateTime> {

    /**
     * Az a dátum- és időformátum amit minden értelmes ember használ
     */
    static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");

    @Override
    public LocalDateTime parse(String text, Locale locale) throws ParseException {
        return LocalDateTime.parse(text, formatter);
    }

    @Override
    public String print(LocalDateTime object, Locale locale) {
        return object.format(formatter);
    }
}

File: src/main/java/com/sem/keeper/formatter/LocalDateTimeFormatter.kt
package com.sem.keeper.formatter

import org.springframework.format.Formatter
import java.text.ParseException
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import java.util.*

class LocalDateTimeFormatter : Formatter<LocalDateTime> {
    @Throws(ParseException::class)
    override fun parse(text: String, locale: Locale): LocalDateTime {
        return LocalDateTime.parse(text, formatter)
    }

    override fun print(`object`: LocalDateTime, locale: Locale): String {
        return `object`.format(formatter)
    }

    companion object {
        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")
    }
}
